sonar-plugin-api/src/main/java/org/sonar/api/server/rule/GenericConceptKeyFormat.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Sonar Plugin API
+ * Copyright (C) 2009-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.server.rule;
+
+/**
+ * The characters allowed for the generic concepts keys are limited by the following regex pattern: {@value GenericConceptKeyFormat#VALID_CHARACTERS_REGEX}
+ */
+public class GenericConceptKeyFormat {
+
+  public static final String ERROR_MESSAGE_SUFFIX = "Generic concept keys accept only the characters: a-z, 0-9, '_'";
+
+  private static final String VALID_CHARACTERS_REGEX = "^[a-z0-9_]+$";
+
+  private static final StringPatternValidator STRING_VALIDATOR = new StringPatternValidator(VALID_CHARACTERS_REGEX, ERROR_MESSAGE_SUFFIX);
+
+  private GenericConceptKeyFormat() {
+    // only static methods
+  }
+
+  public static boolean isValid(String genericConceptKey) {
```

Comment:
This method is not used anywhere. Is the intention here to let the clients use this validator to validate their keys? do we really think there is a use case for that?

Reply:
I think it indeed doesn't make sense to keep this method as it is not used, I have removed it and updated the tests